ql/instruments/capfloor.hpp

```diff
@@ -62,6 +62,15 @@ namespace QuantLib {
                  std::vector<Rate> capRates,
                  std::vector<Rate> floorRates);
         CapFloor(Type type, Leg floatingLeg, const std::vector<Rate>& strikes);
+        CapFloor(Type type,
+                 Leg floatingLeg,
+                 std::vector<Rate> capRates,
+                 std::vector<Rate> floorRates,
+                 bool backwardLooking);
```

Comment:
We can also add `backwardLooking` to the existing constructors and default it to false.

One thing I've been wondering: is this only to support SOFR and the like? If that's the case, we could also try to check somehow the index in the floating leg and avoid the boolean.  The boolean also allows other cases such as a cap on Euribor backward looking, but I don't know if that's something that actually exists...

Reply:
Yes, this is designed solely to support SOFR, ESTR, SONIA, and similar rates. Euribor and the older USD LIBOR cap/floor might have such payoffs (LIBOR In-arrears), which should be considered more exotic, requiring a convexity adjustment for both rates and volatility. (Hopefully we won't have to deal with in-arrear volatility product anymore).

Reply:
Years ago, when I modeled the SOFR cap using ICAP volatility, I used the 3M LIBOR as a proxy index for a 3M daily compounding SOFR cap and combined it with this boolean implementation. After reading your comment, I think simply setting isInArrears_ might achieve the same result.